lib/mlxdevm.c

```diff
@@ -183,6 +183,67 @@ mlxdevm_sf_port_add(struct mlxdevm *dl, uint32_t pfnum, uint32_t sfnum)
 	return NULL;
 }
 
+static int cmd_port_dump_cb_to_list(const struct nlmsghdr *nlh, void *data)
+{
+	struct genlmsghdr *genl = mnl_nlmsg_get_payload(nlh);
+	struct nlattr *tb[MLXDEVM_ATTR_MAX + 1] = {};
+	struct port_list_head *head = data;
+	struct mlxdevm_port_list *cur;
+	struct mlxdevm_port *port;
+
+	mnl_attr_parse(nlh, sizeof(*genl), attr_cb, tb);
+	if (!tb[MLXDEVM_ATTR_DEV_BUS_NAME] || !tb[MLXDEVM_ATTR_DEV_NAME] ||
+	    !tb[MLXDEVM_ATTR_PORT_INDEX] || !tb[MLXDEVM_ATTR_PORT_PCI_PF_NUMBER] ||
+	    !tb[MLXDEVM_ATTR_PORT_PCI_SF_NUMBER])
+		return MNL_CB_ERROR;
+
+	cur = calloc(1, sizeof(struct mlxdevm_port_list));
+	TAILQ_INSERT_TAIL(head, cur, entry);
```

Comment:
do not insert an entry to the list which is not initialized.
this should be after line 208.